### PR TITLE
#97 media types

### PR DESCRIPTION
**Pull request by banukutlu**
Connected to #97 

Note: unrelated change to media types mapping: moved `@formats.compact!` and `@formats.uniq!` out of psulib_config.rb and added to marc_format_processor.rb

### CODE REVIEW

File: .rubocop_todo.yml [hunk 1]
```diff
@@ -1,26 +1,31 @@
...
 
-# Offense count: 3
+# Offense count: 2
+# Configuration parameters: CountComments.
+Metrics/ClassLength:
+  Max: 140
+
+# Offense count: 4
 Metrics/CyclomaticComplexity:
-  Max: 7
```
Comment (banukutlu) [inline .rubocop_todo.yml]: :(

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -1,26 +1,31 @@
...
+  Max: 11
```
Comment (banukutlu) [inline .rubocop_todo.yml]: :(

Comment (banukutlu) [reply .rubocop_todo.yml]: I could not come up with a easy refactor, it could use another set of eyes :)

Comment (cdmo) [reply .rubocop_todo.yml]: Let's make sure to circle back to this one. I don't have a refactor for this one except for pushing every if statement embedded in whens into their own methods.

File: lib/marc_media_type_processor.rb [hunk 3]
```diff
@@ -0,0 +1,181 @@
+# frozen_string_literal: true
+
+# A tool for classifying MARC records using a combination of data from the
+# record's leader and some 949ts to assign media types to records
+class MarcMediaTypeProcessor
```
[the remark this answers is not in the document]
Comment (cdmo) [reply lib/marc_media_type_processor.rb]: Once we have more classes, we can think about this maybe. Just a thought that is starting to form in my head anyway.

[the remark this answers is not in the document]
Comment (banukutlu) [reply lib/marc_media_type_processor.rb]: very good point, I really the parent class `MarcProcessor` idea 👍

**Pull request merged**